lib/logstash/filters/elasticsearch/esql_executor.rb

```diff
@@ -0,0 +1,99 @@
+# encoding: utf-8
+
+module LogStash
+  module Filters
+    class Elasticsearch
+      class EsqlExecutor
+
+        def initialize(plugin, logger)
+          @plugin = plugin
+          @logger = logger
+
+          @query = plugin.params["query"]
+          if @query.strip.start_with?("FROM") && !@query.match?(/\|\s*LIMIT/)
+            @logger.warn("ES|QL query doesn't contain LIMIT, adding `| LIMIT 1` to optimize the performance")
+            @query.concat(' | LIMIT 1')
+          end
+
+          query_params = plugin.params["query_params"] || {}
+          @named_params = query_params["named_params"] || []
+          @fields = plugin.params["fields"]
+          @tag_on_failure = plugin.params["tag_on_failure"]
+          @logger.debug("ES|QL query executor initialized with ", query: @query, named_params: @named_params)
+        end
+
+        def process(client, event)
+          resolved_params = @named_params&.any? ? resolve_parameters(event) : []
+          response = execute_query(client, resolved_params)
+          inform_warning(response)
+          process_response(event, response)
+          @plugin.decorate(event)
+        rescue => e
+          @logger.error("Failed to process ES|QL filter", exception: e)
+          @tag_on_failure.each { |tag| event.tag(tag) }
+        end
+
+        private
+
+        def resolve_parameters(event)
+          @named_params.map do |entry|
+            entry.each_with_object({}) do |(key, value), new_entry|
+              begin
+                resolved_value = event.get(value)
```

Comment:
review note: initially I used `Event#sprintf` (it look output is always `string` but other types can be used in ES|QL query conditions) but after checking details, I think direct access to the event makes sense.